aom/src/main/java/com/nedap/archie/rminfo/RMAttributeInfo.java

```diff
@@ -29,9 +29,14 @@ public RMAttributeInfo(String name, Field field, Class type, Class typeInCollect
         this.setMethod = setMethod;
         this.addMethod = addMethod;
         this.computed = this.setMethod == null && this.addMethod == null;
-        this.isMultipleValued = type instanceof Class && Collection.class.isAssignableFrom(type);
-        this.typeInCollection = typeInCollection;
-        this.typeNameInCollection = typeNameInCollection;
+        this.isMultipleValued = (type instanceof Class && Collection.class.isAssignableFrom(type)) || type.isArray();
+        if(type.isArray()) {
+            this.typeInCollection = type.getComponentType();
+            this.typeNameInCollection = this.typeInCollection.getSimpleName();
+        } else {
+            this.typeInCollection = typeInCollection;
+            this.typeNameInCollection = typeNameInCollection;
+        }
```

Comment:
Maybe this logic should be in the caller? It seems strange that the `typeInCollection` and `typeNameInCollection` are ignored in some special case.

Reply:
Thanks, made much more sense in the caller :)